### PR TITLE
Added support for reply with stream

### PR DESCRIPTION
**Pull request by BobDickinson**
We needed the ability to reply with a stream (as is supported by Nock).  Added that support and tests.  Modified existing replyWithFile to use that mechanism as well (again modeled after how Nock does it).

### CODE REVIEW

File: lib/request.js [hunk 1]
```diff
@@ -233,11 +242,25 @@ Request.prototype.sendResponse = function(response) {
 
   response.writeHead(this.response.statusCode, headers);
 
-  if (this.response.filePath) {
-    var readStream = fs.createReadStream(this.response.filePath);
-    readStream.pipe(response);
+  if (isStream(this.response.body)) {
+    // Because we may need to respond with this body more than once, if it is a stream,
+    // we make a buffer copy and use that as the body for future responses.
```
Comment (kenperkins) [inline lib/request.js]: I'm a little concerned about the implication here if the Stream is massive in size; the copy could easily saturate the memory of a small VM. Obviously Hock is used predominantly for testing, but it's still a concern.

It's possible we could simply not support a stream as the body whenever `Request._maxRequests > 1`.

Thoughts?

Comment (kenperkins) [inline lib/request.js]: The flip side is we already keep a copy of the `body` for any other requests where `_maxRequests > 1`, so in theory this problem already exists for arbitrarily sized bodies.

Maybe we should just take this as is. <ponders>

File: test/many-test.js [hunk 2]
```diff
@@ -249,6 +251,84 @@ describe('Hock Multiple Request Tests', function () {
...
+          request({'url': 'http://localhost:' + PORT + '/url', 'encoding': null}, function (err, res, body) {
+            should.not.exist(err);
+            should.exist(res);
+            res.statusCode.should.equal(200);
+            body.length.should.equal(1000);
+            hockInstance.done(function (err) {
+              should.not.exist(err);
+              done();
+            });
+          });
```
Comment (kenperkins) [inline test/many-test.js]: We should confirm the body matches on successive calls, based on the copy actually matching. I'll do this post merge.

File: lib/request.js (hunk 1 above)
Comment (BobDickinson) [inline lib/request.js]: I added a new PR #26 that only saves the stream contents when actually needed.  So with that, the rule is just "don't use multiple responses for giant stream/files if you don't have room to store a copy of the stream", which is probably reasonable.

**Pull request merged**